src/Admin/Editor.php

```diff
@@ -50,19 +50,37 @@ public function validate_and_pre_save_cb( $data, $post ) {
 		$document = new Document();
 
 		try {
-			if ( array_key_exists( 'post_content', $post ) && 'publish' === $post['post_status'] ) {
-				$data['post_content'] = $document->valid_or_throw( $post['post_content'], $post['ID'] );
+			// Check for empty post_content when publishing the query and throw
+			if ( 'publish' === $post['post_status'] && empty( $post['post_content'] ) ) {
+				throw new RequestError( __( 'Query string is empty', 'wp-graphql-smart-cache' ) );
 			}
+
+			$data['post_content'] = $document->valid_or_throw( $post['post_content'], $post['ID'] );
+
 		} catch ( RequestError $e ) {
 			AdminErrors::add_message( $e->getMessage() );
 
-			// Overwrite new/invalid query with previous working query, or empty
-			$existing_post = get_post( $post['ID'] );
-			if ( $existing_post && property_exists( $existing_post, 'post_content' ) ) {
-				try {
-					$data['post_content'] = $document->valid_or_throw( $existing_post->post_content, $post['ID'] );
-				} catch ( RequestError $e ) {
-					$data['post_content'] = '';
+			// If encountered invalid data when publishing query, revert some data. If draft, allow invalid query.
+			if ( 'publish' === $post['post_status'] ) {
+
+				// If has an existing published post and trying to publish with errors, bail before save_post
+				$existing_post = get_post( $post['ID'], ARRAY_A );
+				if ( $existing_post && 'publish' === $existing_post['post_status'] ) {
+
+					wp_safe_redirect( admin_url( sprintf( '/post.php?post=%d&action=edit', $post['ID'] ) ) );
+					exit;
+
+				} else {
+					$data['post_status'] = 'draft';
+					
+					// This prevents the Admin UI from showing that the post has previously been published (because it actually hasn't been)
+					if ( isset( $existing_post['post_date'] ) ) {
+						$data['post_date']     = $existing_post['post_date'];
+						$data['post_date_gmt'] = get_gmt_from_date( $existing_post['post_date'] );
+					} else {
+						$data['post_date']     = '0000-00-00 00:00:00';
+						$data['post_date_gmt'] = '0000-00-00 00:00:00';
+					}
```

Comment:
```suggestion
					if ( isset( $existing_post['post_date_gmt'] ) && '0000-00-00 00:00:00' !== $existing_post['post_date_gmt'] ) {
						$data['post_date']     = $existing_post['post_date'];
						$data['post_date_gmt'] = get_gmt_from_date( $existing_post['post_date'] );
					} else {
						$data['post_date_gmt'] = '0000-00-00 00:00:00';
					}
```

@markkelnar from my quick testing, I was still seeing invalid documents appear to be published. When I update to this, it seems like a mix of my simple approach and your more guarded approach, but with the intended outcome.

From what I can tell, the post_date can be set to a current timestamp, and this even happens for the ACF validation example, but if `post_date_gmt` is zero'd out, then the UI does not show that the post was published as we were experiencing.

Reply:
I am not able to reproduce. Can you provide more detail about what is happening now?

Reply:
You want to see 'Publish immediately' displayed. Is that what you are considering 'not published'?
![Screenshot 2023-08-15 at 1 23 11 PM](https://github.com/wp-graphql/wp-graphql-smart-cache/assets/749603/b0848dce-6a2e-4d9f-92f5-01163e88ecdc)

Reply:
Ya, that's the same thing that shows when a draft is saved as a draft. When a draft is succesfully published, we see the published date, similar to the screenshot previously shared: 

![CleanShot 2023-08-14 at 08 27 31](https://github.com/wp-graphql/wp-graphql-smart-cache/assets/1260765/11d5da3d-e78a-45ca-bfe1-72127e2450ad)

It creates a funky User Experience if you click "publish" and validation _technically_ prevented publishing, but the UI _appears_ to have published.